src/config.js

```diff
@@ -0,0 +1,1602 @@
+  getSources () {
+    return _.uniq(_.pluck(this.scopedSettingsStore.propertySets, 'source')).sort()
+  }
+
+  // Extended: Retrieve the schema for a specific key path. The schema will tell
+  // you what type the keyPath expects, and other metadata about the config
+  // option.
+  //
+  // * `keyPath` The {String} name of the key.
+  //
+  // Returns an {Object} eg. `{type: 'integer', default: 23, minimum: 1}`.
+  // Returns `null` when the keyPath has no schema specified, but is accessible
+  // from the root schema.
+  getSchema (keyPath) {
+    const keys = splitKeyPath(keyPath)
+    let { schema } = this
+    for (let key of Array.from(keys)) {
+      let childSchema
+      if (schema.type === 'object') {
+        childSchema = schema.properties != null ? schema.properties[key] : undefined
+        if (childSchema == null) {
+          if (isPlainObject(schema.additionalProperties)) {
+            childSchema = schema.additionalProperties
+          } else if (schema.additionalProperties === false) {
+            return null
+          } else {
+            return {type: 'any'}
+          }
+        }
+      } else {
+        return null
+      }
+      schema = childSchema
+    }
+    return schema
+  }
+
+  // Extended: Get the {String} path to the config file being used.
+  getUserConfigPath () {
+    return this.configFilePath
+  }
+
+  // Extended: Suppress calls to handler functions registered with {::onDidChange}
+  // and {::observe} for the duration of `callback`. After `callback` executes,
+  // handlers will be called once if the value for their key-path has changed.
+  //
+  // * `callback` {Function} to execute while suppressing calls to handlers.
+  transact (callback) {
+    this.beginTransaction()
+    try {
+      return callback()
+    } finally {
+      this.endTransaction()
+    }
+  }
+
+  addLegacyScopeAlias (languageId, legacyScopeName) {
+    this.legacyScopeAliases[languageId] = legacyScopeName
+  }
+
+  removeLegacyScopeAlias (languageId) {
+    delete this.legacyScopeAliases[languageId]
+  }
+
+  /*
+  Section: Internal methods used by core
+  */
+
+  // Private: Suppress calls to handler functions registered with {::onDidChange}
+  // and {::observe} for the duration of the {Promise} returned by `callback`.
+  // After the {Promise} is either resolved or rejected, handlers will be called
+  // once if the value for their key-path has changed.
+  //
+  // * `callback` {Function} that returns a {Promise}, which will be executed
+  //   while suppressing calls to handlers.
+  //
+  // Returns a {Promise} that is either resolved or rejected according to the
+  // `{Promise}` returned by `callback`. If `callback` throws an error, a
+  // rejected {Promise} will be returned instead.
+  transactAsync (callback) {
+    let endTransaction
+    this.beginTransaction()
+    try {
+      endTransaction = fn => (...args) => {
+        this.endTransaction()
+        return fn(...Array.from(args || []))
```

Comment:
`args` is always going to be an array here (and therefore non-falsey). this can just be `fn(...args)`